lib/helpers/reporter.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+/**
+ * Build a Pa11y reporter.
+ *
+ * Same as 'pa11y/lib/reporter' but reporter methods accept multiple arguments
+ * @private
+ * @param {Object} methods - The reporter methods.
+ * @returns {Promise} Returns a promise which resolves with the new reporter.
+ */
+module.exports = function buildReporter(methods) {
+	const reporter = {
+		report: new Map(),
+		supports: methods.supports,
+		beforeAll: buildReporterMethod(methods.beforeAll),
+		afterAll: buildReporterMethod(methods.afterAll),
+		begin: buildReporterMethod(methods.begin),
+		results: buildReporterMethod(methods.results),
+		log: {
+			debug: buildReporterMethod(methods.debug),
+			error: buildReporterMethod(methods.error, 'error'),
+			info: buildReporterMethod(methods.info)
+		}
+	};
+
+	return reporter;
+};
+
+/**
+ * Build a Pa11y reporter method, making it async and only outputting when
+ * actual output is returned.
+ * @private
+ * @param {Function} method - The reporter method to build.
+ * @param {String} [consoleMethod='log'] - The console method to use in reporting.
+ * @returns {Function} Returns a built async reporter method.
+ */
+function buildReporterMethod(method, consoleMethod = 'log') {
+	if (typeof method !== 'function') {
+		return () => Promise.resolve();
+	}
```

Comment:
Read the comment about `Promise.all` first. With that in mind then this can be changed to
```
if (typeof method !== 'function`) { return false; }
```

Reply:
Thanks for the feedback. Refactored in https://github.com/pa11y/pa11y-ci/pull/129/commits/36b8016f3fbbead2f9585feb04740c9870e11684